src/components/colorscale/calc.js

```diff
@@ -20,7 +20,7 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
 
     if(containerStr) {
         container = Lib.nestedProperty(trace, containerStr).get();
-        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
+        inputContainer = Lib.nestedProperty(trace, containerStr).get();
```

Comment:
Interesting.

> Can you please give me a use case that would necessitate that the automatically calculated cmin/cmax values need to be persisted on containerIn for the restyle to work out?

I don't remember one off the top of my head. 

**BUT** In any case, changing how we mutate `gd.data` is a backward-incompatible change. For example, what if a user relies on `gd.data[i].marker.cmax` to update a radio button (as in e.g. in the old plot.y/plot workspace) ? So we can't change it before `v2.0.0`.